Firestore/core/test/firebase/firestore/util/status_test.cc

```diff
@@ -113,6 +113,18 @@ TEST(Status, FromErrno) {
           "Already exists: Cannot write file \\(errno .*: File exists\\)"));
 }
 
+TEST(Status, Annotate) {
+  Status a(FirestoreErrorCode::NotFound, "fork not found");
+  Status b = a.Annotate("could not eat dinner");
+  ASSERT_EQ(FirestoreErrorCode::NotFound, b.code());
+  ASSERT_EQ("fork not found; could not eat dinner", b.error_message());
+
+  a = Status(FirestoreErrorCode::NotFound, "");
+  b = a.Annotate("could not eat dinner");
+  ASSERT_EQ(FirestoreErrorCode::NotFound, b.code());
+  ASSERT_EQ("could not eat dinner", b.error_message());
+}
+
```

Comment:
You might also consider adding tests for:

```
TEST(Status, AnnotateOkStatusDoesntAnnotate) {
  a = Status::Ok();
  b = a.Annotate("foo");
  ASSERT_TRUE(b.ok());
  ASSERT_EQ(b.error_message().find("foo") == string::npos);
}
```

Reply:
Will add in the next PR.